cedar-lean/Cedar/Thm/Data/Set.lean

```diff
@@ -280,6 +305,62 @@ theorem inter_wf {α} [LT α] [StrictLT α] [DecidableLT α] [DecidableEq α] {s
   simp only
   apply h₃
 
+theorem union_wf [LT α] [DecidableLT α] [StrictLT α] (s₁ s₂ : Set α) :
+  WellFormed (s₁ ∪ s₂)
+:= by
+  unfold WellFormed
+  simp only [Union.union, union, toList]
+  rw [make_make_eqv]
+  apply List.Equiv.symm
+  exact elts_make_equiv
+
+/-! ### subset -/
+
+theorem elts_subset_then_subset [LT α] [DecidableLT α] [StrictLT α] [DecidableEq α] {xs ys : List α} :
+  xs ⊆ ys → Set.make xs ⊆ Set.make ys
+:= by
+  unfold HasSubset.Subset instHasSubsetSet List.instHasSubsetList subset
+  simp only [List.all_eq_true]
+  intro h₁ x h₂
+  rw [contains_prop_bool_equiv]
+  rw [in_list_iff_in_set] at h₂
+  rw [← make_mem] at *
+  unfold List.Subset at h₁
+  apply h₁ h₂
+
+/--
+  Like `List.subset_def`, but lifted to Sets
+-/
+theorem subset_def [DecidableEq α] {s₁ s₂ : Set α} :
+  s₁ ⊆ s₂ ↔ ∀ a, a ∈ s₁ → a ∈ s₂
+:= by
+  unfold HasSubset.Subset instHasSubsetSet subset
+  simp
```

Comment:
Can you do `simp?` and replace this with the output?

Reply:
I started doing a pass for this in the 3 files touched by this PR, and it's a lot of manual work.  It seems like it could be automated.  Or even rethought: In the event that a Lean upgrade adding something to `simp` breaks a proof, would there be some way to then specifically change to `simp only` just in that place, rather than having to proactively do it everywhere (and also clutter proofs visually)?  It would be awesome if, e.g., in Lean 4.6 we could run some command like `simp4.5?` that would tell us "here's what lemmas `simp` would have used, in Lean 4.5"

Reply:
Seems reasonable to keep separate from this PR then.

I don't think Lean wants to be in the business of managing different versions, but what should work would be to flag the proofs that fail, revert to the previous Lean version, use `simp?` on just those proofs and then do the upgrade. Seems like a nice feature.